desc/coils.py

```diff
@@ -854,6 +882,64 @@ def flip(self, *args, **kwargs):
         """Flip the coils across a plane."""
         [coil.flip(*args, **kwargs) for coil in self.coils]
 
+    def compute_position(self, params=None, source_grid=None, **kwargs):
```

Comment:
I don't think we can easily move this logic into the objective function. We need to be able to compute the objective from an arbitrary `MixedCoilSet`, which could contain a mix of individual coils, coil sets, and other nested mixed coil sets. Only the `CoilSet` can have symmetry, not the `Coil` or `MixedCoilSet` classes. Because of this, it is much easier to have each class compute it's own coil positions. This solution mimics what we already do for `compute_magnetic_field` for each class

Reply:
Don't love this but it works for now and is way less messy than before. Main reason why I dont love is that there are now two different ways to compute the positions for a coilset (compute(x) and compute_position ), even if we make it clear what the differences are and why the latter exists it still would be a point of confusion. If it is not a private function then we should make it clear the use case for it (you want to get all the coil positions possibly accounting for symmetry and NFP, not just unique coils)

Reply:
I would strongly vote for making this a private method for now, since it's only used in the objective and we plan on making a better API in the future. We could also enforce that `source_grid` is only a single grid, the same for all coils.